LiteCore/Query/SQLiteFleeceEach.cc

```diff
@@ -221,8 +224,11 @@ class FleeceCursor : public sqlite3_vtab_cursor {
             Warn("fleece_each filter called with null document! Query is likely to fail. (#379)");
             return SQLITE_OK;
         }
-        data = _vtab->context.delegate->fleeceAccessor(data);
-
+        if (idxNum == kPathIndex) {
+            // If fl_each is called with a 2nd (property path) argument, then the first arg is the
+            // doc body, which we need to extract Fleece from:
+            data = _vtab->context.delegate->fleeceAccessor(data);
+        }
```

Comment:
FYI: This is the fix. The other changes are test cases that Pasin wrote for his earlier PR, or adding/deleting comments.

Reply:
This is the same as what @borrrden has suggested. If fl_each() calls is like `fl_each(fl_nested_value(_V.body, 'sub_dict'), 'array_in_sub_dict')`, the idxNum will be `kFleeceDataIndex`. However I couldn't come up with the SQL that makes translate to the fl_each() call like that.

Reply:
The QueryParser never generates an `fl_each` call like that.

(I know this is sort of fragile, but the better fixes I can think of all involve significant changes/refactoring of the query code. I think we should do that too, but in the meantime this is a decent band-aid.)